### ISSUE TITLE
purefb_policy module fail to update hourly policies

### ISSUE THREAD

**Issue by mahendra-maplelabs** [Pure-Storage-Ansible/FlashBlade-Collection#99]
**Describe the bug**
purefb_policy module failed when try to re-run or update policies < 1day( 86400 sec ) where `at` and `timezone` parameters not required.

**To Reproduce**
1. Run sample playbook
2. Re-run same playbook without any change.
3. See error

**sample playbook**
```
- name: Pure Storage Policy example
  hosts: localhost
  gather_facts: no
  collections:
  - purestorage.flashblade
  tasks:
    - name: Create hourly policy with rules
      purefb_policy:
        name: hourly
        keep_for: 3600  # 1 hour
        every: 3600 # 1 hour
        fb_url: 10.xx.xxx.151
        api_token: T-0b8ad89c-xxxx-xxxx-85ed-286607dc2cd2
```

**Error**
```
TASK [Create hourly policy with rules] *******************************************************************************************
task path: /home/mahendra/Projects/Purestorage/pure-ansible/cleanitup/sample.yml:7
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: mahendra
<127.0.0.1> EXEC /bin/sh -c 'echo ~mahendra && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/mahendra/.ansible/tmp `"&& mkdir "` echo /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950 `" && echo ansible-tmp-1602134847.7509882-22350-260916619499950="` echo /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950 `" ) && sleep 0'
Including module_utils file ansible/__init__.py
Including module_utils file ansible/module_utils/__init__.py
Including module_utils file ansible/module_utils/basic.py
Including module_utils file ansible/module_utils/_text.py
Including module_utils file ansible/module_utils/common/_collections_compat.py
Including module_utils file ansible/module_utils/common/__init__.py
Including module_utils file ansible/module_utils/common/_json_compat.py
Including module_utils file ansible/module_utils/common/_utils.py
Including module_utils file ansible/module_utils/common/file.py
Including module_utils file ansible/module_utils/common/parameters.py
Including module_utils file ansible/module_utils/common/collections.py
Including module_utils file ansible/module_utils/common/process.py
Including module_utils file ansible/module_utils/common/sys_info.py
Including module_utils file ansible/module_utils/common/text/converters.py
Including module_utils file ansible/module_utils/common/text/__init__.py
Including module_utils file ansible/module_utils/common/text/formatters.py
Including module_utils file ansible/module_utils/common/validation.py
Including module_utils file ansible/module_utils/common/warnings.py
Including module_utils file ansible/module_utils/compat/selectors.py
Including module_utils file ansible/module_utils/compat/__init__.py
Including module_utils file ansible/module_utils/compat/_selectors2.py
Including module_utils file ansible/module_utils/distro/__init__.py
Including module_utils file ansible/module_utils/distro/_distro.py
Including module_utils file ansible/module_utils/facts/utils.py
Including module_utils file ansible/module_utils/facts/__init__.py
Including module_utils file ansible/module_utils/facts/compat.py
Including module_utils file ansible/module_utils/facts/ansible_collector.py
Including module_utils file ansible/module_utils/facts/collector.py
Including module_utils file ansible/module_utils/facts/default_collectors.py
Including module_utils file ansible/module_utils/facts/hardware/aix.py
Including module_utils file ansible/module_utils/facts/hardware/__init__.py
Including module_utils file ansible/module_utils/facts/hardware/base.py
Including module_utils file ansible/module_utils/facts/hardware/darwin.py
Including module_utils file ansible/module_utils/facts/hardware/dragonfly.py
Including module_utils file ansible/module_utils/facts/hardware/freebsd.py
Including module_utils file ansible/module_utils/facts/hardware/hpux.py
Including module_utils file ansible/module_utils/facts/hardware/hurd.py
Including module_utils file ansible/module_utils/facts/hardware/linux.py
Including module_utils file ansible/module_utils/facts/hardware/netbsd.py
Including module_utils file ansible/module_utils/facts/hardware/openbsd.py
Including module_utils file ansible/module_utils/facts/hardware/sunos.py
Including module_utils file ansible/module_utils/facts/namespace.py
Including module_utils file ansible/module_utils/facts/network/aix.py
Including module_utils file ansible/module_utils/facts/network/__init__.py
Including module_utils file ansible/module_utils/facts/network/base.py
Including module_utils file ansible/module_utils/facts/network/darwin.py
Including module_utils file ansible/module_utils/facts/network/dragonfly.py
Including module_utils file ansible/module_utils/facts/network/fc_wwn.py
Including module_utils file ansible/module_utils/facts/network/freebsd.py
Including module_utils file ansible/module_utils/facts/network/generic_bsd.py
Including module_utils file ansible/module_utils/facts/network/hpux.py
Including module_utils file ansible/module_utils/facts/network/hurd.py
Including module_utils file ansible/module_utils/facts/network/iscsi.py
Including module_utils file ansible/module_utils/facts/network/linux.py
Including module_utils file ansible/module_utils/facts/network/netbsd.py
Including module_utils file ansible/module_utils/facts/network/nvme.py
Including module_utils file ansible/module_utils/facts/network/openbsd.py
Including module_utils file ansible/module_utils/facts/network/sunos.py
Including module_utils file ansible/module_utils/facts/other/facter.py
Including module_utils file ansible/module_utils/facts/other/__init__.py
Including module_utils file ansible/module_utils/facts/other/ohai.py
Including module_utils file ansible/module_utils/facts/sysctl.py
Including module_utils file ansible/module_utils/facts/system/apparmor.py
Including module_utils file ansible/module_utils/facts/system/__init__.py
Including module_utils file ansible/module_utils/facts/system/caps.py
Including module_utils file ansible/module_utils/facts/system/chroot.py
Including module_utils file ansible/module_utils/facts/system/cmdline.py
Including module_utils file ansible/module_utils/facts/system/date_time.py
Including module_utils file ansible/module_utils/facts/system/distribution.py
Including module_utils file ansible/module_utils/facts/system/dns.py
Including module_utils file ansible/module_utils/facts/system/env.py
Including module_utils file ansible/module_utils/facts/system/fips.py
Including module_utils file ansible/module_utils/facts/system/local.py
Including module_utils file ansible/module_utils/facts/system/lsb.py
Including module_utils file ansible/module_utils/facts/system/pkg_mgr.py
Including module_utils file ansible/module_utils/facts/system/platform.py
Including module_utils file ansible/module_utils/facts/system/python.py
Including module_utils file ansible/module_utils/facts/system/selinux.py
Including module_utils file ansible/module_utils/facts/system/service_mgr.py
Including module_utils file ansible/module_utils/facts/system/ssh_pub_keys.py
Including module_utils file ansible/module_utils/facts/system/user.py
Including module_utils file ansible/module_utils/facts/timeout.py
Including module_utils file ansible/module_utils/facts/virtual/base.py
Including module_utils file ansible/module_utils/facts/virtual/__init__.py
Including module_utils file ansible/module_utils/facts/virtual/dragonfly.py
Including module_utils file ansible/module_utils/facts/virtual/freebsd.py
Including module_utils file ansible/module_utils/facts/virtual/hpux.py
Including module_utils file ansible/module_utils/facts/virtual/linux.py
Including module_utils file ansible/module_utils/facts/virtual/netbsd.py
Including module_utils file ansible/module_utils/facts/virtual/openbsd.py
Including module_utils file ansible/module_utils/facts/virtual/sunos.py
Including module_utils file ansible/module_utils/facts/virtual/sysctl.py
Including module_utils file ansible/module_utils/parsing/convert_bool.py
Including module_utils file ansible/module_utils/parsing/__init__.py
Including module_utils file ansible/module_utils/pycompat24.py
Including module_utils file ansible/module_utils/six/__init__.py
Including module_utils file ansible_collections/purestorage/flashblade/plugins/module_utils/purefb.py
Including module_utils file ansible_collections/__init__.py
Including module_utils file ansible_collections/purestorage/__init__.py
Including module_utils file ansible_collections/purestorage/flashblade/__init__.py
Including module_utils file ansible_collections/purestorage/flashblade/plugins/__init__.py
Including module_utils file ansible_collections/purestorage/flashblade/plugins/module_utils/__init__.py
Using module file /home/mahendra/.ansible/collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_policy.py
<127.0.0.1> PUT /home/mahendra/.ansible/tmp/ansible-local-22300kd2e5kyd/tmpdoxbyvga TO /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950/AnsiballZ_purefb_policy.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950/ /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950/AnsiballZ_purefb_policy.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/mahendra/.pyenv/versions/3.6.9/bin/python3.6 /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950/AnsiballZ_purefb_policy.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/mahendra/.ansible/tmp/ansible-tmp-1602134847.7509882-22350-260916619499950/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/tmp/ansible_purefb_policy_payload_6f6g5wf3/ansible_purefb_policy_payload.zip/ansible_collections/purestorage/flashblade/plugins/modules/purefb_policy.py", line 373, in update_policy
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/purity_fb_1dot9/apis/policies_api.py", line 1602, in update_policies
    (data) = self.update_policies_with_http_info(policy_patch, **kwargs)
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/purity_fb_1dot9/apis/policies_api.py", line 1686, in update_policies_with_http_info
    collection_formats=collection_formats)
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/api_client/api_client.py", line 331, in call_api
    _return_http_data_only, collection_formats, _preload_content, _request_timeout)
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/api_client/api_client.py", line 156, in __call_api
    _request_timeout=_request_timeout)
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/api_client/api_client.py", line 392, in request
    body=body)
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/api_client/rest.py", line 299, in PATCH
    body=body)
  File "/home/mahendra/.pyenv/versions/3.6.9/lib/python3.6/site-packages/purity_fb/api_client/rest.py", line 235, in request
    raise ApiException(http_resp=r)
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "at": null,
            "enabled": true,
            "every": 3600,
            "fb_url": "10.xx.xxx.151",
            "filesystem": null,
            "keep_for": 3600,
            "name": "hourly",
            "replica_link": null,
            "state": "present",
            "timezone": "Asia/Kolkata"
        }
    },
    "msg": "Failed to update policy hourly."
}

PLAY RECAP ***********************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
```

**Expected behavior**
purefb_policy module should not fail in re-run.

**Platform versions:**
purity-fb version v1.9.2
collection - latest code
Ansible version 2.10

**Comment by sdodsley** [comment]
Addressed by #101

**Comment by mahendra-maplelabs** [comment]
Thanks @sdodsley , #101  resolved the issue.
